src/Razor/src/Microsoft.CodeAnalysis.Razor/EventHandlerTagHelperDescriptorProvider.cs

```diff
@@ -83,11 +83,21 @@ private List<EventHandlerData> GetEventHandlerData(Compilation compilation)
 
                     if (Equals(attribute.AttributeClass, eventHandlerAttribute))
                     {
+                        var enablePreventDefault = false;
+                        var enableStopPropagation = false;
+                        if (attribute.ConstructorArguments.Length == 4)
```

Comment:
Should flow the Razor version to the `TagHelperDescriptor` providers as part of the context. This is a bit too fragile for future updates. Basically by the end of all of these changes i'd imagine 0 `TagHelperDescriptor` provider changes.

Reply:
Okay I'll try using the RazorVersion here. 

> Basically by the end of all of these changes i'd imagine 0 TagHelperDescriptor provider changes.

I'm confused. There will still be changes to this file to add preventDefault and stopPropagation as bound attribute parameters conditionally based on the RazorVersion.

Reply:
Yup sorry you're right. There will be that 😁

Reply:
I'm  ok with surfacing the version here, I'm also OK with not. I think it's fine to say that the runtime declares its support for this by including it in the attributes.

Reply:
Keep in mind that flowing the razor version around could require VS changes as well.

Reply:
> Keep in mind that flowing the razor version around could require VS changes as well.

Yup, in the Razor LanguageServer as well. 

I think flowing version is the more future proof change though. I imagine us continuing to add different Blazor TagHelper capabilities; of which those could benefit from efforts that we've done at the parser layer to ensure backwards compatibility. All that being said I'm definitely a little biased because I view these two features as core concepts to what an "event" is capable of.

Reply:
I'm just not sure that *version* is the right concept. An non-HTML-DOM event wouldn't support these features.

Reply:
> I'm just not sure that _version_ is the right concept

And I totally get this because a RazorLanguageVersion of 3.1 could potentially run on more than just web, meaning there needs to be some other sort of "version" type thing that dictates what the runtime is capable of. 

I hate to rely on attribute constructor argument inspection for that though because for each platform Blazor supports we'll then need to do one of the following:

1. Add a TagHelper feature flag control of some sort that the TagHelperDescriptorProvider code can use to turn features on and off. This would be a lot like the feature capabilities we have at the parser layer.
2. Add heuristics that can detect which type of runtime we're operating on by looking at the types involved. This couples the TagHelperDescriptor discovery phase to each potential runtime and its shape.
3. Have every runtimes C# attributes look the same. This means for non-web projects their EventHandlerAttribute C# classes will also have a preventDefault attribute even if it's not applicable to the runtime involved. This would definitely lead to tricky situations where our runtime types get polluted with every possible runtimes feature capability

With how this PR currently looks we're going down path 2 or 3 which i'm not totally thrilled with. 

I propose the path we go down is 1. It'd introduce a feature capability set that we would currently infer from RazorLanguageVersion; that would give us the flexibility in the future to untie it from RazorLangaugeVersion when new Blazor runtimes are introduced.

Reply:
> Add a TagHelper feature flag control of some sort that the TagHelperDescriptorProvider code can use to turn features on and off. This would be a lot like the feature capabilities we have at the parser layer.

@NTaylorMullen, this does seem elegant but I feel like there are a few blockers. How would we figure out what features to turn on or off? The runtime will no longer have any control over this. What layer/based on what will this be decided?

Reply:
> How would we figure out what features to turn on or off?
> What layer/based on what will this be decided?

"It'd introduce a feature capability set that we would currently infer from RazorLanguageVersion"



> The runtime will no longer have any control over this.

You're right, it wont directly but that's how Razor works anyhow. It uses an inferred language version / configuration to generate C# that "should" work.

Reply:
> "It'd introduce a feature capability set that we would currently infer from RazorLanguageVersion"
> You're right, it wont directly but that's how Razor works anyhow. It uses an inferred language version / configuration to generate C# that "should" work.

But given @rynowak's [comment](https://github.com/aspnet/AspNetCore-Tooling/pull/1217#discussion_r333538563) above, we can't make that decision based on the language version because non-Html DOM event?